### ISSUE TITLE
Exit scheduler if no clients connected

### ISSUE THREAD

**Issue by jacobtomlinson** [dask/distributed#2346]
Is there a way to have the scheduler process exit automatically if there are no connected clients for some period of time?

**Comment by jacobtomlinson** [comment]
The motivation is avoiding accidentally leaving large clusters hanging around on fargate. If I create a remote scheduler and ton of workers, connect to them and use them for a bit, then close my laptop and go home without shutting them down they stay around forever. 

My initial thought was something along the lines of:

```python
try:
    create_cluster()
    do_stuff()
finally:
    delete_cluster()
```

However this is not safe from my Python process being killed.

**Comment by mrocklin** [comment]
Yeah, I can see how that would make sense.  We do something similar for workers today with the `--death-timeout` keyword (added in https://github.com/dask/distributed/pull/940)

I can imagine implementing this internally in one of two ways:

1.  Whenever we remove our last client (in the `Scheduler.remove_client` method) we would start a coroutine that would wait until it was either woken up by a `Condition` or until the specified time had passed.  If the specified time had passed, then it would close the Scheduler.  Whenever a new client was added (see `Scheduler.add_client`) it would notify that coroutine by notifying the condition, which would be placed as a private variable on the scheduler.
2.  We start a periodic callback that checks every second for the presence of a client.  If enough seconds pass without having seen one, it closes the scheduler.  This is simpler to implement but prone to failure in situations where clients show up briefly to make short requests.  

    We could also look at `scheduler.events['all']` for `add-client` events, though depending on logging information like this seems slightly more brittle to me.

I'm inclined to go for option 1.  I think that it would make a good task for someone who wanted to gain more experience with async programming and futzing about within the Dask scheduler.  Any interest @jacobtomlinson ?

**Comment by jacobtomlinson** [comment]
This sounds perfect, I'm using the death timeout to close all the workers in my fargate clusters so similar functionality in the scheduler is great. I'll add it to my very long to-do list, but if someone else wants to pick it up in the mean time then great!

**Comment by mrocklin** [comment]
It would also be helpful to add a timeout if a client is connected, but no one has done any work in a long while.  I would use this today when I leave Dask clusters up and running on shared resources. 

I'll try to work this up later this week

FYI @quasiben
